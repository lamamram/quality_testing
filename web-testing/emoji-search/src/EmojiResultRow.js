import React from 'react';
import PropTypes from 'prop-types';
import './EmojiResultRow.css';

const EmojiResultRow = ({ title, symbol }) => {
  const codePointHex = symbol.codePointAt(0).toString(16);
  const src = `${process.env.PUBLIC_URL}/emoji-32/${codePointHex}.png`;
  return (
    <div
      className="component-emoji-result-row"
      data-clipboard-text={symbol}
    >
      <img alt={title} src={src}/>
      <span className="title">{title}</span>
      <span className="info">Click to copy emoji</span>
    </div>
  );
};

EmojiResultRow.propTypes = {
  title: PropTypes.string,
  symbol: PropTypes.string
};
export default EmojiResultRow;
