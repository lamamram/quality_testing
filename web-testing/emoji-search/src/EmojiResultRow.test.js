import React from 'react';
import EmojiResultRow from './EmojiResultRow';
import { render } from '@testing-library/react';

describe('the EmojiResultRow component', () => {
  test('should render the title correctly', () => {
    const { getByText } = render(<EmojiResultRow title="smile" symbol="☺" />);

    const titleElement = getByText('smile');
    expect(titleElement).toBeInTheDocument();
  });

  test('should render the component via snapshot', () => {
    const { container } = render(<EmojiResultRow title="smile" symbol="☺" />);
    expect(container).toMatchSnapshot();
  });
});