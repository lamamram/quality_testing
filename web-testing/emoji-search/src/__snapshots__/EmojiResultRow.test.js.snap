// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the EmojiResultRow component should render the component via snapshot 1`] = `
<div>
  <div
    class="component-emoji-result-row copy-to-clipboard"
    data-clipboard-text="â˜º"
  >
    <img
      alt="smile"
      src="/emoji-32/263a.png"
    />
    <span
      class="title"
    >
      smile
    </span>
    <span
      class="info"
    >
      Click to copy emoji
    </span>
  </div>
</div>
`;
